Mobile Sales Explorartory Data Analysis (Python Project)

Import the necessary libraries

In [2]:
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd

Load the Dataset

In [3]:
data = pd.read_excel("Mobile Sales Data for Python Project.xlsx", sheet_name="Phone Sales")
data.head()

Date      Country     Distributor  Brand Operator  Unit Cost  Amount  \
0 2018-05-24     Colombia           Metro  Apple   Tuenti        750      89   
1 2020-05-31  El Salvador  Saga Falabella  Apple   Tuenti        750      34   
2 2021-08-30  Switzerland           Metro  Apple   Tuenti        750      18   
3 2021-08-03       Mexico         Oeschle  Apple   Tuenti        750     129   
4 2018-03-21     Paraguay          Ripley  Apple   Tuenti        750      95   

   Unit Price   Sales  
0        1200  106800  
1        1200   40800  
2        1200   21600  
3        1200  154800  
4        1200  114000

In [4]:
data.tail()

Date    Country     Distributor    Brand  Operator  Unit Cost  \
2993 2020-08-23    Germany          Tottus  Samsung  Movistar        590   
2994 2018-08-17       Cuba  Saga Falabella  Samsung  Movistar        590   
2995 2018-12-13    Denmark          Ripley  Samsung  Movistar        590   
2996 2019-12-09    Germany          Ripley  Samsung  Movistar        590   
2997 2020-04-14  Venezuela         Oeschle  Samsung  Movistar        590   

      Amount  Unit Price  Sales  
2993      44         820  36080  
2994       5         820   4100  
2995      17         820  13940  
2996       7         820   5740  
2997      37         820  30340

Basic information about Dataset

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2998 entries, 0 to 2997
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   Date         2998 non-null   datetime64[ns]
 1   Country      2998 non-null   object        
 2   Distributor  2998 non-null   object        
 3   Brand        2998 non-null   object        
 4   Operator     2998 non-null   object        
 5   Unit Cost    2998 non-null   int64         
 6   Amount       2998 non-null   int64         
 7   Unit Price   2998 non-null   int64         
 8   Sales        2998 non-null   int64         
dtypes: datetime64[ns](1), int64(4), object(4)
memory usage: 210.9+ KB


=> there are 2998(0 to 2997) rows and total 9 columns 
    4 numeric columns(Amount, Unit Cost, Unit Price, Sales) and 4 more columns(Country, Distributor, Brand, Opeartor) and a date column

Statistical summary of the Dataset

In [6]:
data.describe()

Date    Unit Cost       Amount   Unit Price  \
count                           2998  2998.000000  2998.000000  2998.000000   
mean   2020-04-08 15:12:36.504336128   541.534356    50.815210   830.126751   
min              2018-01-03 00:00:00   175.000000     1.000000   450.000000   
25%              2019-03-16 06:00:00   550.000000    14.000000   800.000000   
50%              2020-05-30 00:00:00   590.000000    33.000000   820.000000   
75%              2021-05-12 00:00:00   675.000000    70.000000   950.000000   
max              2021-12-31 00:00:00   750.000000   295.000000  1200.000000   
std                              NaN   169.081053    52.102526   210.767344   

               Sales  
count    2998.000000  
mean    42429.886591  
min       450.000000  
25%     10862.500000  
50%     26000.000000  
75%     57000.000000  
max    336000.000000  
std     46234.737273

=> According to above information, Minimum unit cost = 175
                                Maximum unit cost = 750
                                Min Ammount = 1
                                Max Ammount = 295
                                Min Unit Price = 450
                                Max unit Price = 1200
                                Min Sale = 450
                                Max Sale = 336000

To see the missing values in the DataSet

In [7]:
data.isnull()

Date  Country  Distributor  Brand  Operator  Unit Cost  Amount  \
0     False    False        False  False     False      False   False   
1     False    False        False  False     False      False   False   
2     False    False        False  False     False      False   False   
3     False    False        False  False     False      False   False   
4     False    False        False  False     False      False   False   
...     ...      ...          ...    ...       ...        ...     ...   
2993  False    False        False  False     False      False   False   
2994  False    False        False  False     False      False   False   
2995  False    False        False  False     False      False   False   
2996  False    False        False  False     False      False   False   
2997  False    False        False  False     False      False   False   

      Unit Price  Sales  
0          False  False  
1          False  False  
2          False  False  
3          False  False  
4          False  False  
...          ...    ...  
2993       False  False  
2994       False  False  
2995       False  False  
2996       False  False  
2997       False  False  

[2998 rows x 9 columns]

In [8]:
data.isnull().sum()

Date           0
Country        0
Distributor    0
Brand          0
Operator       0
Unit Cost      0
Amount         0
Unit Price     0
Sales          0
dtype: int64

=> Means there is no missing values in any column

To see the shape of the dataset

In [9]:
data.shape

(2998, 9)

To see the duplicate rows

In [10]:
data.duplicated().sum()

np.int64(1499)

=> This dataset has 1499 duplicate rows so we need to remove them

In [11]:
data = data.drop_duplicates().reset_index(drop=True)

In [12]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1499 entries, 0 to 1498
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   Date         1499 non-null   datetime64[ns]
 1   Country      1499 non-null   object        
 2   Distributor  1499 non-null   object        
 3   Brand        1499 non-null   object        
 4   Operator     1499 non-null   object        
 5   Unit Cost    1499 non-null   int64         
 6   Amount       1499 non-null   int64         
 7   Unit Price   1499 non-null   int64         
 8   Sales        1499 non-null   int64         
dtypes: datetime64[ns](1), int64(4), object(4)
memory usage: 105.5+ KB


In [13]:
data.tail()

Date    Country     Distributor    Brand  Operator  Unit Cost  \
1494 2020-08-23    Germany          Tottus  Samsung  Movistar        590   
1495 2018-08-17       Cuba  Saga Falabella  Samsung  Movistar        590   
1496 2018-12-13    Denmark          Ripley  Samsung  Movistar        590   
1497 2019-12-09    Germany          Ripley  Samsung  Movistar        590   
1498 2020-04-14  Venezuela         Oeschle  Samsung  Movistar        590   

      Amount  Unit Price  Sales  
1494      44         820  36080  
1495       5         820   4100  
1496      17         820  13940  
1497       7         820   5740  
1498      37         820  30340

After removal of duplicate data now we have 1499 entries 0 to 1498

In [14]:
data.isnull().sum()

Date           0
Country        0
Distributor    0
Brand          0
Operator       0
Unit Cost      0
Amount         0
Unit Price     0
Sales          0
dtype: int64

To check datatype

In [15]:
data.dtypes

Date           datetime64[ns]
Country                object
Distributor            object
Brand                  object
Operator               object
Unit Cost               int64
Amount                  int64
Unit Price              int64
Sales                   int64
dtype: object

In [16]:
print(data['Date'].dtypes)

datetime64[ns]


Dataset is in correct format

In [17]:
data

Date      Country     Distributor    Brand  Operator  Unit Cost  \
0    2018-05-24     Colombia           Metro    Apple    Tuenti        750   
1    2020-05-31  El Salvador  Saga Falabella    Apple    Tuenti        750   
2    2021-08-30  Switzerland           Metro    Apple    Tuenti        750   
3    2021-08-03       Mexico         Oeschle    Apple    Tuenti        750   
4    2018-03-21     Paraguay          Ripley    Apple    Tuenti        750   
...         ...          ...             ...      ...       ...        ...   
1494 2020-08-23      Germany          Tottus  Samsung  Movistar        590   
1495 2018-08-17         Cuba  Saga Falabella  Samsung  Movistar        590   
1496 2018-12-13      Denmark          Ripley  Samsung  Movistar        590   
1497 2019-12-09      Germany          Ripley  Samsung  Movistar        590   
1498 2020-04-14    Venezuela         Oeschle  Samsung  Movistar        590   

      Amount  Unit Price   Sales  
0         89        1200  106800  
1         34        1200   40800  
2         18        1200   21600  
3        129        1200  154800  
4         95        1200  114000  
...      ...         ...     ...  
1494      44         820   36080  
1495       5         820    4100  
1496      17         820   13940  
1497       7         820    5740  
1498      37         820   30340  

[1499 rows x 9 columns]

Make a YearMonth column by extracting the year and month from the Date column

In [18]:
data['YearMonth'] = data['Date'].dt.to_period("M")
data.head()

Date      Country     Distributor  Brand Operator  Unit Cost  Amount  \
0 2018-05-24     Colombia           Metro  Apple   Tuenti        750      89   
1 2020-05-31  El Salvador  Saga Falabella  Apple   Tuenti        750      34   
2 2021-08-30  Switzerland           Metro  Apple   Tuenti        750      18   
3 2021-08-03       Mexico         Oeschle  Apple   Tuenti        750     129   
4 2018-03-21     Paraguay          Ripley  Apple   Tuenti        750      95   

   Unit Price   Sales YearMonth  
0        1200  106800   2018-05  
1        1200   40800   2020-05  
2        1200   21600   2021-08  
3        1200  154800   2021-08  
4        1200  114000   2018-03